lib/ash/type/type.ex

```diff
@@ -535,13 +535,15 @@ defmodule Ash.Type do
   end
 
   defp do_generator({:array, type}, constraints) do
-    generator = do_generator(type, item_constraints(constraints))
+    item_constraints = item_constraints(constraints)
+    generator = do_generator(type, item_constraints)
 
     generator =
       if constraints[:nil_items?] do
         StreamData.one_of([StreamData.constant(nil), generator])
       else
         generator
+        |> StreamData.filter(&(Ash.Type.apply_constraints(type, &1, item_constraints) != nil))
```

Comment:
🤔 `apply_constraints/3` returns `{:ok, value} | :error | {:error, error}`

Reply:
Sorry, my bad. I just ran the test and assumed it would work.